generate/templates/partials/field_accessors.cc

```diff
@@ -258,7 +261,7 @@
         else {
           // promise was rejected
           baton->result = {{ field.return.error }};
-          baton->done = false;
+          baton->done = true;
```

Comment:
I don't remember why we did this originally but this change makes a lot more sense now. :+1: